### PR TITLE
[ENH] Fixes in Catch22

### LINKED ISSUE #1384: [ENH] Update Catch22 transformer
### Describe the feature or idea you want to propose

Since its implementation, the numba version of `Catch22` ([code here](https://github.com/aeon-toolkit/aeon/blob/main/aeon/transformations/collection/feature_based/_catch22.py)) has become slightly out of date compared to the [original implementation](https://github.com/DynamicsAndNeuralSystems/pycatch22).

The features which now differ should be updated, and parity with the original output should be verified

### Describe your proposed solution

Look at the [authors implementation](https://github.com/DynamicsAndNeuralSystems/pycatch22) (C code), and update our version of the features in numba to match.

The transformer has a parameter to use this package, so the results can be compared to verify they are the same (or similar).

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Moonzyyy**
#### Reference Issues/PRs
Fixes #1384

#### What does this implement/fix? Explain your changes.
Looked through the original C code implementation found in #1384, and made sure code does not cause issues with Numba.

#### Does your contribution introduce a new dependency? If yes, which one?
None

#### Any other comments?
This has not been tested to see if it has fixed #1598. More testing required. From my understanding of messing around with Numba however, it may be worth creating our own np.argsort.

### CODE REVIEW

File: aeon/transformations/collection/feature_based/_catch22.py [hunk 1]
```diff
@@ -1315,3 +1371,161 @@ def _verify_features(features, catch24):
...
+    return sd
+
+
+@njit(fastmath=True, cache=True)
+def _autocorr_lag(x, size, lag):
+    return _corr(x, x[lag:], size - lag)
+
+
+@njit(fastmath=True, cache=True)
+def _corr(x, y, size):
```
[the remark this answers is not in the document]
Comment (Moonzyyy) [reply aeon/transformations/collection/feature_based/_catch22.py:1457]: > there is a bit of overhead when compiling numba functions, so good to keep the amount as low as possibly. Is corr only used in autocorr_lag?

meaning if a function is only used once I should remove it and put it where its called? If so yes, I can do that for some of the new functions implemented

[the remark this answers is not in the document]
Comment (MatthewMiddlehurst) [reply aeon/transformations/collection/feature_based/_catch22.py:1457]: Only if it makes sense to do so really. I will leave it up to you to decide if its worth it. Not a blocke.

**Comment by Moonzyyy** [comment]
Here are the result files.

Old:
[aeon_catch22_with_numba.xlsx](https://github.com/user-attachments/files/16321778/aeon_catch22_with_numba.xlsx)

New:
[aeon_catch22_with_numba.xlsx](https://github.com/user-attachments/files/16321774/aeon_catch22_with_numba.xlsx)

Compare To pycatch22:
[pycatch22_catch22_ipd.xlsx](https://github.com/user-attachments/files/16321775/pycatch22_catch22_ipd.xlsx)

File: aeon/transformations/collection/feature_based/_catch22.py [hunk 2]
```diff
@@ -696,46 +742,53 @@ def _SC_FluctAnal_2_rsrangefit_50_1_logi_prop_r1(X):
     @staticmethod
     @njit(fastmath=True, cache=True)
     def _SB_TransitionMatrix_3ac_sumdiagcov(X, acfz):
-        # Trace of covariance of transition matrix between symbols in 3-letter
-        # alphabet.
-        ds = np.zeros(int((len(X) - 1) / acfz + 1))
+
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/collection/feature_based/_catch22.py:745]: extra white space

**Review by MatthewMiddlehurst — COMMENTED** [review]
Could you change `_verify_features` to allow usage of the shortened features and then add a test for this?

File: aeon/regression/tests/test_all_regressors.py [hunk 3]
```diff
@@ -89,30 +89,30 @@ def test_regressor_against_expected_results(self, estimator_class):
...
+    #         inst.predict(X)
+
+    # def test_does_not_override_final_methods(self, estimator_class):
+    #     """Test does not override final methods."""
+    #     if "fit" in estimator_class.__dict__:
+    #         raise ValueError(f"Classifier {estimator_class} overrides the method fit")
+    #     if "predict" in estimator_class.__dict__:
+    #         raise ValueError(
+    #             f"Classifier {estimator_class} overrides the method " f"predict"
+    #         )
```
Comment (MatthewMiddlehurst) [inline aeon/regression/tests/test_all_regressors.py:118]: Don't comment these out

**Comment by MatthewMiddlehurst** [comment]
division by zero in previous notebook test is a bit worrying, no idea what function caused it though.
https://github.com/aeon-toolkit/aeon/actions/runs/10159947289/job/28095237710?pr=1809

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**